evaluate/repository.go

```diff
@@ -80,3 +83,86 @@ func Repository(logger *log.Logger, resultPath string, model evalmodel.Model, la
 
 	return repositoryAssessment, problems, nil
 }
+
+// TemporaryRepository creates a temporary repository and initializes a git repo in it.
+func TemporaryRepository(logger *log.Logger, dataPath string) (temporaryRepositoryPath string, cleanup func(), err error) {
+	temporaryPath, err := os.MkdirTemp("", "eval-dev-quality")
```

Comment:
Why not handling the error right way and only after defining the `cleanup` function?

Reply:
ah nice, that was a ctrl+c/ctrl+v mistake of course.